File: RemoveRecursionRefactoring/tests/ro/pub/cs/diploma/RemoveRecursionFixTest.java
package ro.pub.cs.diploma;

/*
 * @see com.siyeh.ig.fixes.performance.RemoveTailRecursionFixTest
 */
public class RemoveRecursionFixTest extends IGQuickFixesTestCase {
  @Override
  protected void setUp() throws Exception {
    super.setUp();
    myFixture.enableInspections(new RemoveRecursionInspection());
    myRelativePath = "fixes";
    myDefaultHint = RemoveRecursionBundle.INSTANCE.message("remove.recursion.replace.quickfix");
  }

  public void testCallOnOtherInstance1() {
    doTest();
  }

  public void testCallOnOtherInstance2() {
    doTest();
  }

  public void testDependency1() {
    doTest();
  }

  public void testDependency2() {
    doTest();
  }

  public void testDependency3() {
    doTest();
  }

  public void testDependency4() {
    doTest();
  }

  public void testThisVariable() {
    doTest();
  }

  public void testUnmodifiedParameter() {
    doTest();
  }

  public void testArrayPrinter1() {
    doTest();
  }

  public void testArrayPrinter2() {
    doTest();
  }

  public void testArrayPrinter3() {
    doTest();
  }

  public void testFib1() {
    doTest();
  }

  public void testFib2() {
    doTest();
  }

  public void testFactorial1() {
    doTest();
  }

  public void testFactorial2() {
    doTest();
  }

  public void testBreakStatement() {
    doTest();
  }

  public void testLabeledBreakStatement() {
    doTest();
  }

  public void testNameClash() {
    doTest();
  }

  public void testContinueStatement() {
    doTest();
  }

  public void testDoWhileStatement() {
    doTest();
  }

  public void testP1() {
    doTest();
  }

  public void testLocalVariableSameName() {
    doTest();
  }

  public void testMergesort() {
    doTest();
  }

  public void testForeachArray() {
    doTest();
  }

  public void testBreakStatement1() {
    doTest();
  }

  public void testContinueStatement1() {
    doTest();
  }

  public void testFibSwitch() {
    doTest();
  }

  public void testFibSwitchFallThrough() {
    doTest();
  }
}

File: inspection/tests/ro/pub/cs/diploma/RemoveRecursionFixTest.kt
package ro.pub.cs.diploma

/*
 * @see com.siyeh.ig.fixes.performance.RemoveTailRecursionFixTest
 */
class RemoveRecursionFixTest : IGQuickFixesTestCase() {
  @Throws(Exception::class)
  override fun setUp() {
    super.setUp()
    myFixture.enableInspections(RemoveRecursionInspection())
    myRelativePath = "fixes"
    myDefaultHint = RemoveRecursionBundle.message("remove.recursion.replace.quickfix")
  }

  fun testCallOnOtherInstance1() {
    doTest()
  }

  fun testCallOnOtherInstance2() {
    doTest()
  }

  fun testDependency1() {
    doTest()
  }

  fun testDependency2() {
    doTest()
  }

  fun testDependency3() {
    doTest()
  }

  fun testDependency4() {
    doTest()
  }

  fun testThisVariable() {
    doTest()
  }

  fun testUnmodifiedParameter() {
    doTest()
  }

  fun testArrayPrinter1() {
    doTest()
  }

  fun testArrayPrinter2() {
    doTest()
  }

  fun testArrayPrinter3() {
    doTest()
  }

  fun testFib1() {
    doTest()
  }

  fun testFib2() {
    doTest()
  }

  fun testFactorial1() {
    doTest()
  }

  fun testFactorial2() {
    doTest()
  }

  fun testBreakStatement() {
    doTest()
  }

  fun testLabeledBreakStatement() {
    doTest()
  }

  fun testNameClash() {
    doTest()
  }

  fun testContinueStatement() {
    doTest()
  }

  fun testDoWhileStatement() {
    doTest()
  }

  fun testP1() {
    doTest()
  }

  fun testLocalVariableSameName() {
    doTest()
  }

  fun testMergesort() {
    doTest()
  }

  fun testForeachArray() {
    doTest()
  }

  fun testBreakStatement1() {
    doTest()
  }

  fun testContinueStatement1() {
    doTest()
  }

  fun testFibSwitch() {
    doTest()
  }

  fun testFibSwitchFallThrough() {
    doTest()
  }
}
